### DISCUSSION TITLE
Question: Ignoring file subset in commit diff

### DISCUSSION THREAD

**Discussion by BumbarG** [magit/magit#4704] [Support] [answered]
Hello,
is it possible to exclude a file subset from a commit diff (e.g. directory with vendored code)? So an inverse to the `Limit to files` command.

**Comment by tarsius** [comment, marked as answer]
It's possible, see below.

~~I'm afraid `git diff` does not offer an option for that.~~

**Reply by kyleam** [reply]
What about using the `exclude` magic signature in the pathspec?

```
$ git diff --name-only 5a7519fd^-
docs/magit.org
docs/magit.texi
lisp/magit-ediff.el

$ git diff --name-only 5a7519fd^- -- :^docs
lisp/magit-ediff.el
```

When Magit inserts the diff, it drops the `--literal-pathspecs` that's usually in `magit-git-global-arguments`, so passing ":^path/to/exclude" as the "Limit to files:" value should work.

**Reply by BumbarG** [reply]
Great, that works! Thank you both for the quick responses.

**Reply by tarsius** [reply]
Doh, didn't you already tell me about this a little while ago?

In reverse news, your reply should have been an "answer" in its own right, not a "comment" on my wrong answer. Stackoverflow like. :stuck_out_tongue_closed_eyes:
